Runtime/FQDN new tests

eloycoto opened the pull request:
New FQDN test based on a local bind9 container that uses three new containers in a different network to be able to continue the work over different DNS types without hit external DNS.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/6289)
<!-- Reviewable:end -->

eloycoto committed:
Test: Helpers add docker run options on SampleContainers.

This change allow us to add a new options in the `docker run` for sample
containers.

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

eloycoto committed:
Test: Add runtime/fqdn.go test.

This new test enables a new way to test FQDN policies. It creates a new
bind9 server with a custom conf on domain cilium.test.

Bind9 container uses three new containers in a different docker network
(For cilium is world identity) so can be used without internet.

Also migrated current fqdn policy test to the new one.

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

aanm commented on test/runtime/fqdn.go:
```diff
@@ -0,0 +1,248 @@
+
+var _ = Describe("RuntimeFQDNPolicies", func() {
+	const (
+		bindContainerImage = "eloycoto/docker-bind"
```
add this to the cilium dockerhub, specify a version and write the repository where this is installed. `docker.io/...`

eloycoto committed:
Runtime: Use cilium/docker-bind container.

Use new docker-bind container from Cilium organization.
Use docker.io and the label in the docker image.

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

aanm reviewed (commented):
I'll defer the test code to @raybejjani
